addons/mail/static/src/models/discuss/discuss.js

```diff
@@ -4,128 +4,48 @@ import { registerModel } from '@mail/model/model_core';
-                // Channel must be pinned immediately to be able to open it before
-                // the result of join is received on the bus.
-                channel.update({ isServerPinned: true });
-                channel.open();
-            }
-        },
-        /**
-         * @param {Object} req
-         * @param {string} req.term
-         * @param {function} res
-         */
-        async handleAddChannelAutocompleteSource(req, res) {
-            this.update({ addingChannelValue: req.term });
-            const threads = await this.messaging.models['Thread'].searchChannelsToOpen({ limit: 10, searchTerm: req.term });
-            const items = threads.map((thread) => {
-                const escapedName = escape(thread.name);
-                return {
-                    id: thread.id,
-                    label: escapedName,
-                    value: escapedName,
-                };
-            });
-            const escapedValue = escape(req.term);
-            // XDU FIXME could use a component but be careful with owl's
-            // renderToString https://github.com/odoo/owl/issues/708
-            items.push({
-                label: _.str.sprintf(
-                    `<strong>${this.env._t('Create %s')}</strong>`,
-                    `<em><span class="fa fa-hashtag"/>${escapedValue}</em>`,
-                ),
-                escapedValue,
-                special: 'public'
-            }, {
-                label: _.str.sprintf(
-                    `<strong>${this.env._t('Create %s')}</strong>`,
-                    `<em><span class="fa fa-lock"/>${escapedValue}</em>`,
-                ),
-                escapedValue,
-                special: 'private'
-            });
-            res(items);
-        },
-        /**
-         * @param {Event} ev
-         * @param {Object} ui
-         * @param {Object} ui.item
-         * @param {integer} ui.item.id
-         */
-        handleAddChatAutocompleteSelect(ev, ui) {
-            this.messaging.openChat({ partnerId: ui.item.id });
-            this.clearIsAddingItem();
-        },
-        /**
-         * @param {Object} req
-         * @param {string} req.term
-         * @param {function} res
-         */
-        handleAddChatAutocompleteSource(req, res) {
-            
         },
         /**
          * Handles click on the mobile "new chat" button.
          *
          * @param {MouseEvent} ev
          */
         onClickMobileNewChatButton(ev) {
-            this.update({ isAddingChat: true });
+            this.view.update({
+                mobileAddChatInputView: (
+                    this.view.mobileAddChatInputView
+                    ? clear()
+                    : insertAndReplace({ doFocus: true })
+                ),
+            });
         },
         /**
          * Handles click on the mobile "new channel" button.
          *
          * @param {MouseEvent} ev
          */
         onClickMobileNewChannelButton(ev) {
```

Comment:
handlers should be moved to vies (eventually)

Reply:
Makes sense too, but I wasn't sure if you would accept to turn all discuss accessors to discussView in discuss components immediately